From d36c5e4a9b90446fb2ef6f7b7712176c4b732b32 Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Wed, 11 Jul 2018 11:49:19 -0700
Subject: [PATCH 10/70] Inform user to turn off APM for
 IMS_SIP_ALTERNATE_EMERGENCY_CALL cause.

When IMS_SIP_ALTERNATE_EMERGENCY_CALL disconnect cause is received,
inform the user to turn off airplane mode in order to make a call.

Test: make -j 64; no manual testing possible.
Bug: 69024173
Change-Id: I76afa4796f1b7ad2b9595bc8680559fa9e83d462
---
 .../android/services/telephony/DisconnectCauseUtil.java  | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/com/android/services/telephony/DisconnectCauseUtil.java b/src/com/android/services/telephony/DisconnectCauseUtil.java
index dd0f41604..f725c4926 100644
--- a/src/com/android/services/telephony/DisconnectCauseUtil.java
+++ b/src/com/android/services/telephony/DisconnectCauseUtil.java
@@ -155,6 +155,7 @@ public class DisconnectCauseUtil {
             case android.telephony.DisconnectCause.IMEI_NOT_ACCEPTED:
             case android.telephony.DisconnectCause.WIFI_LOST:
             case android.telephony.DisconnectCause.IMS_ACCESS_BLOCKED:
+            case android.telephony.DisconnectCause.IMS_SIP_ALTERNATE_EMERGENCY_CALL:
                 return DisconnectCause.ERROR;
 
             case android.telephony.DisconnectCause.DIALED_MMI:
@@ -289,6 +290,10 @@ public class DisconnectCauseUtil {
                 resourceId = R.string.callFailed_data_limit_reached;
                 break;
 
+            case android.telephony.DisconnectCause.IMS_SIP_ALTERNATE_EMERGENCY_CALL:
+                resourceId = R.string.incall_error_power_off;
+                break;
+
             default:
                 break;
         }
@@ -653,6 +658,10 @@ public class DisconnectCauseUtil {
                 resourceId = R.string.callFailed_wifi_lost;
                 break;
 
+            case android.telephony.DisconnectCause.IMS_SIP_ALTERNATE_EMERGENCY_CALL:
+                resourceId = R.string.incall_error_power_off;
+                break;
+
             default:
                 break;
         }
-- 
2.17.1

